Tests/test_builtinfunc.py

```diff
@@ -308,17 +310,118 @@ def test_unichr(self):
         self.assertTrue(unichr(0) == '\x00')
         self.assertTrue(unichr(max_uni) == max_ok_value)
 
-    def test_max_min(self):
+    def test_max(self):
+        self.assertEqual(max('123123'), '3')
         self.assertTrue(max([1,2,3]) == 3)
         self.assertTrue(max((1,2,3)) == 3)
         self.assertTrue(max(1,2,3) == 3)
-
-        self.assertTrue(min([1,2,3]) == 1)
-        self.assertTrue(min((1,2,3)) == 1)
-        self.assertTrue(min(1,2,3) == 1)
+        self.assertTrue(max((1, 2, 3, 1, 2, 3)) == 3)
+        self.assertEqual(max([1, 2, 3, 1, 2, 3]), 3)
+        self.assertTrue(max(1,2,3) == 3)
+        self.assertTrue(max([], default=1) == 1)
+        self.assertTrue(max((), default=1) == 1)
+        self.assertTrue(max([], default=None) is None)
 
         self.assertEqual(max((1,2), None), (1, 2))
-        self.assertEqual(min((1,2), None), None)
+
+        self.assertEqual(max(1, 2, 3.0), 3.0)
+        self.assertEqual(max(1, 2.0, 3), 3)
+        self.assertEqual(max(1.0, 2, 3), 3)
+
+        self.assertRaises(TypeError, max)
+        self.assertRaises(TypeError, max, 42)
+        self.assertRaises(ValueError, max, ())
+        class BadSeq:
+            def __getitem__(self, index):
+                raise ValueError
+        self.assertRaises(ValueError, max, BadSeq())
+
+        for stmt in (
+            "max(key=int)",                 # no args
+            "max(default=None)",
+            "max(1, 2, default=None)",      # require container for default
+            "max(default=None, key=int)",
+            "max(1, key=int)",              # single arg not iterable
+            "max(1, 2, keystone=int)",      # wrong keyword
+            "max(1, 2, key=int, abc=int)",  # two many keywords
+            "max(1, 2, key=1)",             # keyfunc is not callable
+            ):
+            try:
+                exec(stmt, globals())
+            except TypeError:
+                pass
+            else:
+                self.fail(stmt)
+
+        self.assertEqual(max((1,), key=neg), 1)     # one elem iterable
+        self.assertEqual(max((1,2), key=neg), 1)    # two elem iterable
+        self.assertEqual(max(1, 2, key=neg), 1)     # two elems
+        self.assertEqual(max((), default=None), None)    # zero elem iterable
+        #self.assertEqual(min((1,), default=None), 1)     # one elem iterable
```

Comment:
Any reason why these cases are commented out?

Reply:
I am receiving TypeError: NoneType is not callable from
object retValue = PythonCalls.Call(context, method, args[retIndex]);
Not sure how to deal with these tests cases

Reply:
Alright, this one is a bit tricky. The previous code assumed that if you got to the `Call` then it was because `key` was a keyword argument. However in this case we only have `default` and `key` is undefined. It'll require a bit more work to get this working. Off the top of my head you could do something like:
```C#
public static object max(CodeContext/*!*/ context, object x, [ParamDictionary]IDictionary<object, object> dict) {
    // ...

    if (!i.MoveNext()) {
        if (def != UndefinedKeywordArgument) return def;
        throw PythonOps.ValueError("max() arg is an empty sequence");
    }

    if (method == UndefinedKeywordArgument) {
        return max(context, x);
    }

    // ...
}

private static object UndefinedKeywordArgument = new object();

private static Tuple<object, object> VerifyKeys(string name, IDictionary<object, object> dict) {
    object method;
    object def;

    int cnt = 0;
    if (dict.TryGetValue("key", out method)) {
        cnt++;
    } else {
        method = UndefinedKeywordArgument;
    }

    if (dict.TryGetValue("default", out def)) {
        cnt++;
    } else {
        def = UndefinedKeywordArgument;
    }

    // ...
}
```

Or something else so you can know if it's undefined...